juju/testing/apiserver.go

```diff
@@ -0,0 +1,586 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/clock/testclock"
+	"github.com/juju/cmd/v3"
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	mgotesting "github.com/juju/mgo/v3/testing"
+	"github.com/juju/names/v4"
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/utils/v3"
+	"github.com/prometheus/client_golang/prometheus"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/api"
+	"github.com/juju/juju/apiserver"
+	"github.com/juju/juju/apiserver/apiserverhttp"
+	"github.com/juju/juju/apiserver/authentication/macaroon"
+	"github.com/juju/juju/apiserver/observer"
+	"github.com/juju/juju/apiserver/observer/fakeobserver"
+	"github.com/juju/juju/apiserver/stateauthenticator"
+	apitesting "github.com/juju/juju/apiserver/testing"
+	"github.com/juju/juju/cloud"
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/core/auditlog"
+	coredatabase "github.com/juju/juju/core/database"
+	corelogger "github.com/juju/juju/core/logger"
+	"github.com/juju/juju/core/multiwatcher"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/core/presence"
+	"github.com/juju/juju/database/testing"
+	"github.com/juju/juju/jujuclient"
+	"github.com/juju/juju/mongo"
+	"github.com/juju/juju/mongo/mongotest"
+	"github.com/juju/juju/pubsub/centralhub"
+	"github.com/juju/juju/state"
+	coretesting "github.com/juju/juju/testing"
+	"github.com/juju/juju/testing/factory"
+	"github.com/juju/juju/worker/lease"
+	wmultiwatcher "github.com/juju/juju/worker/multiwatcher"
+)
+
+const AdminSecret = "dummy-secret"
+
+var (
+	// AdminUser is the default admin user for a controller.
+	AdminUser = names.NewUserTag("admin")
+
+	// DefaultCloudRegion is the default cloud region for a controller model.
+	DefaultCloudRegion = "dummy-region"
+
+	// DefaultCloud is the default cloud for a controller model.
+	DefaultCloud = cloud.Cloud{
+		Name:      "dummy",
+		Type:      "ec2",
```

Comment:
would be good to not use a real provider name here

Reply:
I tried it but too much stuff still looks for a cloud called "dummy"
Will revisit once jujuconnsuite is gone